python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1496,6 +1503,26 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
 
         exposureMetadata["LSST ISR NOMINAL PTC USED"] = nominalPtcUsed
 
+        # If the user inputs an older PTC, we need to check if we need to apply
+        # a bug fix.
+        if not nominalPtcUsed:
+            # The user is inputting a PTC.
+            ptcVersion = ptc.getMetadata()['PTC_VERSION']
+            if ptcVersion < 1.9 and ptc.ptcFitType == "FULLCOVARIANCE":
+                # Before version 1.9, the noise stored in the PTC was in units
+                # of electron^2 only if ptcFitType == FULLCOVARIANCE. After
+                # version 1.9, we standardized the PhotonTransferCurveDataset
+                # noise units to electron to fix this bug. If a user tries to
+                # use an earlier version of PTC with this fit type, we must be
+                # sure to do the calculations properly. More information about
+                # this noise issue can be found in DM-45976.
+                self.log.info(f"Input PTC VERSION ({ptcVersion}) < 1.9 and"
+                              "ptcFitType == FULLCOVARIANCE. Applying fix"
+                              f"for DM-45976 noise issue.")
+                for amp in detector:
+                    ptc.noise[amp.getName()] = numpy.sqrt(ptc.noise[amp.getName()])
+
+        # Set which gains to use
```

Comment:
Similar to above, if this code is in the `ptcDataset` then it won't need to be repeated (with the same formatting errors 😄 ).